### ISSUE TITLE
Kodi 19 RC crashes on Mac OS while scrolling on file list

### ISSUE THREAD

**Issue by jaroslawhartman** [xbmc/xbmc#19208]
### Describe the bug
Here is a clear and concise description of what the problem is:

When scrolling through a file list in Kodi 19 RC (Mac OS), the app keeps crashing.

Crash log: https://paste.kodi.tv/vugapebavi.kodi

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

No crash.

## Actual Behavior
Crash as shown in https://paste.kodi.tv/vugapebavi.kodi

### To Reproduce
1. Browse to a SMB folder
2. Scroll through the list, either with mouse or keyboard
3.

### Debuglog

The debuglog can be found here:

Debug https://paste.kodi.tv/kutipagofo.kodi

### Screenshots 
Here are some links or screenshots to help explain the problem:
N/A

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:
N/A

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [ ] tvOS
 - [ ] Linux
 - [X] OSX
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: macOS 11.2 (20D64)
 - Kodi version: 19.0.rc1 (20210115-90a1e12804)

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by kambala-decapitator** [comment]
FYI can't reproduce on Catalina.

1. does it happen in 100% of cases?
2. do you have any custom binary addons installed?

**Comment by jaroslawhartman** [comment]
Hi,

1. Yes, it happens all the time when listing this particular _NFS_ share. I'm guessing this might be cause by a special character in one of the file/folder names. See https://youtu.be/3KXcveIOEUU for screen recording.
2. Content of addons folder:

```
drwxr-xr-x   9 jhartman  staff   288 Jul  6  2019 script.revolve.service
drwxr-xr-x   9 jhartman  staff   288 Jul  6  2019 service.skin.widgets
drwxr-xr-x   8 jhartman  staff   256 Jul  6  2019 resource.images.recordlabels.white
drwxr-xr-x   8 jhartman  staff   256 Jul  6  2019 resource.images.moviegenreicons.coloured
drwxr-xr-x   8 jhartman  staff   256 Jul  6  2019 resource.images.musicgenreicons.poster
drwxr-xr-x   8 jhartman  staff   256 Jul  6  2019 resource.images.weathericons.animated
drwxr-xr-x   8 jhartman  staff   256 Jul  6  2019 resource.images.weatherfanart.prairie
drwxr-xr-x   8 jhartman  staff   256 Jul  6  2019 resource.images.weatherfanart.multi
drwxr-xr-x   5 jhartman  staff   160 Dec 29  2019 metadata.common.theaudiodb.com
drwxr-xr-x   8 jhartman  staff   256 Dec 29  2019 script.grab.fanart
drwxr-xr-x  12 jhartman  staff   384 Dec 29  2019 skin.transparency
drwxr-xr-x  11 jhartman  staff   352 Mar 25  2020 service.subtitles.napiprojekt
drwxr-xr-x   5 jhartman  staff   160 Aug 22 13:25 metadata.common.allmusic.com
drwxr-xr-x   5 jhartman  staff   160 Aug 22 13:25 metadata.common.musicbrainz.org
drwxr-xr-x   5 jhartman  staff   160 Aug 22 13:25 metadata.common.fanart.tv
drwxr-xr-x   7 jhartman  staff   224 Aug 22 13:25 metadata.artists.universal
drwxr-xr-x   6 jhartman  staff   192 Dec 26 22:10 metadata.common.imdb.com
drwxr-xr-x   8 jhartman  staff   256 Dec 26 22:10 resource.images.studios.white
drwxr-xr-x   7 jhartman  staff   224 Jan 10 14:26 metadata.themoviedb.org
drwxr-xr-x   5 jhartman  staff   160 Jan 10 14:26 metadata.common.themoviedb.org
drwxr-xr-x  13 jhartman  staff   416 Jan 10 14:26 skin.revolve
drwxr-xr-x   8 jhartman  staff   256 Jan 13 15:54 metadata.tvshows.themoviedb.org
drwxr-xr-x   7 jhartman  staff   224 Jan 13 16:01 metadata.themoviedb.org.python
drwxr-xr-x   5 jhartman  staff   160 Jan 13 16:01 resource.language.pl_pl
drwxr-xr-x   9 jhartman  staff   288 Jan 13 16:01 script.favourites
drwxr-xr-x   9 jhartman  staff   288 Jan 13 16:01 script.globalsearch
drwxr-xr-x   9 jhartman  staff   288 Jan 13 16:01 script.image.resource.select
drwxr-xr-x   7 jhartman  staff   224 Jan 13 16:01 script.module.soupsieve
drwxr-xr-x  10 jhartman  staff   320 Jan 13 16:01 script.module.beautifulsoup4
drwxr-xr-x   7 jhartman  staff   224 Jan 13 16:01 script.module.certifi
drwxr-xr-x   8 jhartman  staff   256 Jan 13 16:01 script.module.chardet
drwxr-xr-x   6 jhartman  staff   192 Jan 13 16:01 script.module.future
drwxr-xr-x   8 jhartman  staff   256 Jan 13 16:01 script.module.idna
drwxr-xr-x   6 jhartman  staff   192 Jan 13 16:01 script.module.kodi-six
drwxr-xr-x   7 jhartman  staff   224 Jan 13 16:01 script.module.routing
drwxr-xr-x   8 jhartman  staff   256 Jan 13 16:01 script.module.simplejson
drwxr-xr-x   6 jhartman  staff   192 Jan 13 16:01 script.module.unidecode
drwxr-xr-x   8 jhartman  staff   256 Jan 13 16:01 script.module.urllib3
drwxr-xr-x   6 jhartman  staff   192 Jan 13 16:01 script.module.simpleeval
drwxr-xr-x   9 jhartman  staff   288 Jan 13 16:01 script.skinshortcuts
drwxr-xr-x   8 jhartman  staff   256 Jan 13 16:01 script.module.requests
drwxr-xr-x   8 jhartman  staff   256 Jan 13 16:01 service.library.data.provider
drwxr-xr-x  10 jhartman  staff   320 Jan 13 16:01 service.subtitles.napisy24pl
drwxr-xr-x   9 jhartman  staff   288 Jan 13 16:01 service.subtitles.opensubtitles
drwxr-xr-x  14 jhartman  staff   448 Jan 13 16:02 skin.aeon.nox.silvo
drwxr-xr-x   9 jhartman  staff   288 Jan 13 16:02 weather.gismeteo
drwxr-xr-x   8 jhartman  staff   256 Jan 13 16:12 script.module.defusedxml
drwxr-xr-x   8 jhartman  staff   256 Jan 13 16:12 script.keymap
drwxr-xr-x   7 jhartman  staff   224 Feb  6 12:13 metadata.album.universal
drwxr-xr-x   8 jhartman  staff   256 Feb  6 12:13 metadata.tvshows.themoviedb.org.python
drwxr-xr-x  56 jhartman  staff  1792 Feb  6 12:13 packages
drwxr-xr-x   6 jhartman  staff   192 Feb  6 12:13 service.xbmc.versioncheck
drwxr-xr-x   2 jhartman  staff    64 Feb  6 12:13 temp
```

**Comment by jaroslawhartman** [comment]
With a bit of try-and-error I found an avi file causing this problem. This is **not** an issue caused by the filename but rather the content.

Troublemaking file stored in https://wieloryb.uk.to/own/index.php/s/2fk7Ghk0gTQmjSq . I will share the pwd separately.

Best regards,
Jarek

**Comment by kambala-decapitator** [comment]
> it happens all the time when listing this particular NFS share

in the initial message you wrote that it's SMB. What is correct?

I've tested your failing movie on 2 machines using v19RC1:
1. Mojave, access over SMB - no crash
2. Catalina, access to local folder (I've added it as SMB share to my .local domain) - no crash

I don't have access to Big Sur atm (can try to get one in the next days). Could you check if it's NFS issue specifically?

**Comment by jaroslawhartman** [comment]
Hi, with more triage - it's crashing even on a local folder.

**Comment by kambala-decapitator** [comment]
tested on Big Sur on ARM Mac with local file - no crash...

I've configured folder without any specific type of content, it has only a single video file in it. Is it sufficient or I need lots of files to make the list scrollable?

**Comment by ksooo** [comment]
Tested on macOS 11.1, x86_64 - cannot reproduce

**Comment by jaroslawhartman** [comment]
Hi, 
As I wrote, it was sufficient to have this single corrupted file (which I've shared directly with @kambala-decapitator ). Placement of the file doesn't matter - it could be a local disk or a remote share.

Cobsistentley same crash, on BigSur Intel and Kodi 18&19.

https://youtu.be/3KXcveIOEUU

**Comment by kambala-decapitator** [comment]
try this workaround by removing code signature:

    codesign -fs - /path/to/kodi.app

**Comment by jaroslawhartman** [comment]
Hi @kambala-decapitator 

> `codesign --remove-signature /path/to/kodi.app`

Thanks, seems this workaround stops crashes. Do you think this issue could be fixed? 

Best regards,
Jarek

**Comment by kambala-decapitator** [comment]
I've no idea what is the exact reason and no hints from the internet.

@jaroslawhartman Could you try to test/debug a [nightly build](https://mirrors.kodi.tv/nightlies/osx/x86_64/master/)? It's also signed, but has debugging allowed, hopefully we'll get more info at the moment of crash.

**Comment by jaroslawhartman** [comment]
Hi,

Tests on a nightly build:

https://paste.kodi.tv/imeyudiliz.kodi
https://paste.kodi.tv/xupumezegi.kodi

Best regards,
Jarek

**Comment by kambala-decapitator** [comment]
Thanks.

just noticed that you have external subs:

> 2021-02-23 15:20:41.845 T:12610773    INFO <general>: ScanPathsForAssociatedItems: found associated file smb://WIELORYB/torrents/done3/Pantani The Accidental Death Of A Cyclist 2014.en.srt

maybe they're causing the crash? Could you upload this file?

**Comment by jaroslawhartman** [comment]
Hi @kambala-decapitator 

I've deleted all other files from the video folder (including subtitles). Still same crashes.

Best regards, Jarek

**Comment by fuzzard** [comment]
Can you disable thumbnail extraction

Settings needs to be Standard or higher

Settings->Media->Extract Thumbnail from Videos

**Comment by fabquenneville** [comment]
Hello I also have issues of Kodi crashing sometimes when opening menus and NFS filesystems involved. If anyone wants logs feel free to contact me.

**Comment by Fred-IV** [comment]
This fixed it! Even 19.4 has this bug.
I had deleted and rebuilt my library so many times.
I was in the process of setting up Plex to replace Kodi because I was so sick of it.
Plex really sucks though, it doesn't really work for local media. It still makes you log into their server, for your own local files???? And what happens if the internet goes down?
I'm really glad I finally found this solution.
It seems like it's been an issue for a long time. Kodi really needs to fix this bug, or disable that feature by default.

**Comment by fritsch** [comment]
Should be fixed by: https://github.com/xbmc/xbmc/pull/21599 in Nexus and by https://github.com/xbmc/xbmc/pull/21630 backported to Matrix. If a 19.5 is released the fix is in there. Until then either install Nexus nightly or Matrix nightly build.

**Comment by stephanecharette** [comment]
I understand this issue says it was fixed and closed, but I was having this problem with v18 and v19 on my Mac Mini 2018.  I'm now on Kodi v19.4 and it still crashes shortly after startup, or seconds after I try to access "Movies".

I went into Settings -> Media -> Videos and turned off "Extract video information from files" as well as "Extract chapter thumbnails", and now it runs perfectly.  No more crash.  Thanks for the hint.
